datafusion/physical-plan/src/execution_plan.rs

```diff
@@ -110,6 +110,16 @@ pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
     /// trait, which is implemented for all `ExecutionPlan`s.
     fn properties(&self) -> &PlanProperties;
 
+    /// Returns an error if this individual node does not conform to its invariants.
```

Comment:
Perhaps to take into account the different types of "executableness" we can use a similar enum as we did for LogicalPlans: https://github.com/apache/datafusion/blob/264f4c51fc97981435f1a1827de934472d60edf8/datafusion/expr/src/logical_plan/invariants.rs#L31


Then the signature might look like
```rust
    fn check_node_invariants(&self, invariant_level: InvariantLevel) -> Result<()> 
      Ok(())
    }
```

Reply:
It felts like we had "executableness" defined in two ways: (1) general plan validity which is already encoded in the SanityPlanCheck, and (2) any per ExecutionPlan node invariants. For the latter case (the function signature above) we could define an InvariantLevel. Given that the scope of this PR has shifting a bit to [implementing this approach](https://github.com/apache/datafusion/pull/13986#discussion_r1914279888) , do we still want to use the invariant levels?